server/vcs/provider/github/file_fetcher.go

```diff
@@ -15,28 +15,44 @@ type RemoteFileFetcher struct {
 	ClientCreator githubapp.ClientCreator
 }
 
-func (r *RemoteFileFetcher) GetModifiedFilesFromCommit(ctx context.Context, repo models.Repo, sha string, installationToken int64) ([]string, error) {
+type FileFetcherOptions struct {
+	Sha   string
+	PRNum int
+}
+
+func (r *RemoteFileFetcher) GetModifiedFiles(ctx context.Context, repo models.Repo, installationToken int64, fileFetcherOptions FileFetcherOptions) ([]string, error) {
+	client, err := r.ClientCreator.NewInstallationClient(installationToken)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating installation client")
+	}
+
+	var fileFetcher func(ctx context.Context, client *gh.Client, repo models.Repo, fileFetcherOptions FileFetcherOptions, listOptions gh.ListOptions) ([]*gh.CommitFile, *gh.Response, error)
```

Comment:
nit: in general i like keep this function types simple 

```go
func(ctx context.Context) ([]*gh.CommitFile, *gh.Response, error)
```

and just wrapping within this function itself.